public/src/Main.purs

```diff
@@ -6,18 +6,42 @@ import Effect.Aff (Aff, launchAff_)
 import Halogen as H
 import Halogen.Aff as HA
 import Halogen.HTML as HH
+import Halogen.HTML.CSS as HCSS
 import Halogen.VDom.Driver (runUI)
 import Component.Router as Router
 import Routing.Hash (getHash, matchesWith)
 import Routing.Duplex (parse)
-import Halogen (liftAff, liftEffect)
-import Model.Route (Route, routeCodec)
+import Halogen (liftEffect)
+
 import Data.Either (hush)
 import Data.Maybe (Maybe(..))
-import Effect.Ref (Ref)
+-- import Effect.Ref (Ref) // 
 import Effect.Ref as Ref
+import Data.Foldable (traverse_) -- used to mount style
+import Data.Const (Const)
+import Data.Newtype (unwrap)
+import Web.DOM.ParentNode (QuerySelector(..)) -- for style to get the head element
+
+import Model.Route (routeCodec)
+import Model.AppEnv (AppEnv, runAppM)
+import CSS as CSS
+import CSS.Stylesheet (CSS)
+-- import CSS.Color as COLOR
+import CSS ((?))
 
-import Model.AppEnv (AppEnv, runAppM, LogLevel, ApiUrl, BaseUrl)
+styleComponent :: CSS -> forall m. H.Component HH.HTML (Const Void) Unit Void m
```

Comment:
@Benjmhart can you explain what this style component does? If I understand it correctly it creates a Halogen component, which is an object, with initial state, a render method that attaches a stylesheet, an eval method that shouldn't get called (absurd???), and a receiver that always evaluates to the nothing side of a maybe. Correct me if I am wrong.